tests/components/fan/test_dyson.py

```diff
@@ -325,9 +379,309 @@ def test_service_set_night_mode(self):
         self.hass.services.call(dyson.DOMAIN, dyson.SERVICE_SET_NIGHT_MODE,
                                 {"entity_id": "fan.bed_room",
                                  "night_mode": True}, True)
-        assert not dyson_device.night_mode.called
+        assert not dyson_device.set_night_mode.called
 
         self.hass.services.call(dyson.DOMAIN, dyson.SERVICE_SET_NIGHT_MODE,
                                 {"entity_id": "fan.living_room",
                                  "night_mode": True}, True)
-        dyson_device.night_mode.assert_called_with(True)
+        dyson_device.set_night_mode.assert_called_with(True)
+
+    @mock.patch('libpurecool.dyson.DysonAccount.devices',
+                return_value=[_get_purecool_device_on()])
+    @mock.patch('libpurecool.dyson.DysonAccount.login', return_value=True)
+    def test_async_purecool_added_to_hass(self, mocked_login, mocked_devices):
+        """Test async added to hass."""
+        setup_component(self.hass, dyson_parent.DOMAIN, {
+            dyson_parent.DOMAIN: {
+                dyson_parent.CONF_USERNAME: "email",
+                dyson_parent.CONF_PASSWORD: "password",
+                dyson_parent.CONF_LANGUAGE: "US",
+                }
+            })
+        self.hass.block_till_done()
+        assert len(self.hass.data[dyson.DYSON_DEVICES]) == 1
+        assert mocked_devices.return_value[0].add_message_listener.called
+
+    def test_dyson_purecool_turn_on(self):
+        """Test turn on purecool fan."""
+        device = _get_purecool_device_on()
+        component = dyson.DysonPureCoolDevice(self.hass, device)
```

Comment:
This is not a component, it's an entity.

Please don't test the entity in isolation. Set up the component and let that set up the corresponding platform, which adds the entity. Then assert that the core state machine has the correct state for the entity.

Do this while patching appropriately to avoid I/O.